xbmc/addons/AddonInfo.h

```diff
@@ -85,13 +85,240 @@ namespace ADDON
+     */
+    const std::string& Summary() const { return m_summary; }
+
+    /*!
+     * @brief To get a bigger add-on description
+     *
+     * @return description text
+     */
+    const std::string& Description() const { return m_description; }
+
+    /*!
+     * @brief To get the master library name of an add-on
+     *
+     * Currently is this used only for binary add-ons where only one library is
+     * present where everything becomes included.
+     *
+     * @return the library name of add-on
+     */
+    const std::string& LibName() const { return m_libname; }
+
+    /*!
+     * @brief to get the author of add-on
+     * @return Author name
+     */
+    const std::string& Author() const { return m_author; }
+
+    /*!
+     * @brief The source URL of the add-on
+     * @return Source URL where add-on is from
+     */
+    const std::string& Source() const { return m_source; }
+
+    /*!
+     * @brief Path where add-on is present
+     *
+     * Can be local or also repository URL's
+     *
+     * @return Path where add-on is
+     */
+    const std::string& Path() const { return m_path; }
+
+    /*!
+    * @brief Set the Path where add-on is present
+    *
+    * Can be local or also repository URL's
+    *
+    * @param[in] path path where addon is present
+    */
+    void SetPath(const std::string& path) { m_path = path; }
+
+    /*!
+     * @brief To get a changelog from add-on
+     *
+     * Used to inform the user about changes
+     *
+     * @return Changelog text
+     */
+    const std::string& ChangeLog() const { return m_changelog; }
+
+    /*!
+     * @brief To get Path with icon file
+     *
+     * @return If a icon present a path to them, otherwise empty
+     */
+    const std::string& Icon() const { return m_icon; }
+
+    /*!
+     * @brief To get a add-on fanart picture
+     *
+     * Is used as background on kodi's add-on window.
+     *
+     * @return Path with image who used as fanart
+     */
+    const ArtMap& Art() const { return m_art; }
+
+    /*!
+     * @brief To get a list of available screenshots from add-on
+     *
+     * Used on kodi's add-on window to descripe a bit from add-on
+     *
+     * @return a list of screenshots if present
+     */
+    const std::vector<std::string>& Screenshots() const { return m_screenshots; }
+
+    /*!
+     * @brief To get a disclaimer of warranties
+     *
+     * @return a disclaimer of addon
+     */
+    const std::string& Disclaimer() const { return m_disclaimer; }
+
+    /*!
+     * @brief To get a list with dependencies related to this add-on
+     */
+    const ADDONDEPS& GetDeps() const { return m_dependencies; }
+
+    /*!
+     * @brief To get broken parts
+     *
+     * @todo make description more detailed
+     */
+    const std::string& Broken() const { return m_broken; }
+
+    /*!
+     * @brief To get install date of add-on
+     *
+     * @note this is only be set from add-on browser and not always present!
```

Comment:
That's not true. It should be set everywhere except when getting a single addon from addon manager. (And that is just a workaround due to some parts calling GetAddon way too much)